ibm_cloud_sdk_core/jwt_token_manager.py

```diff
@@ -51,151 +48,15 @@ class JWTTokenManager:
-        the past 60 seconds), and if so will sleep for a short time interval
-        (currently 0.5 seconds) before checking again. The check for an active
-        request and update of `request_time` are serailized by the `lock`
-        variable so that only one caller can become the active requester
-        with a 60 second interval.
-
-        Threads that sleep waiting for the active request to complete will
-        eventually find a newly valid token and return, or 60 seconds will
-        elapse and a new thread will assume the role of the active request.
-        """
-        while self._is_token_expired():
-            current_time = self._get_current_time()
-
-            with self.lock:
-                request_active = self.request_time > (current_time - 60)
-                if not request_active:
-                    self.request_time = current_time
-
-            if not request_active:
-                token_response = self.request_token()
-                self._save_token_info(token_response)
-                self.request_time = 0
-                return
-
-            time.sleep(0.5)  # Sleep for 0.5 seconds before checking token again
-
-
-    def request_token(self) -> None:
-        """Should be overridden by child classes.
-
-        Raises:
-            NotImplementedError: Thrown when called.
-        """
-        raise NotImplementedError(
-            'request_token MUST be overridden by a subclass of JWTTokenManager.'
-        )
-
-    @staticmethod
-    def _get_current_time() -> int:
-        return int(time.time())
-
-    def _is_token_expired(self) -> bool:
-        """
-        Check if currently stored token is expired.
-
-        Returns
-        -------
-        bool
-            True if token is expired; False otherwise
-        """
-        current_time = self._get_current_time()
-        return self.expire_time < current_time
-
-    def _token_needs_refresh(self) -> bool:
-        """
-        Check if currently stored token needs refresh.
-
-        Returns
-        -------
-        bool
-            True if token needs refresh; False otherwise
-        """
-        current_time = self._get_current_time()
-
-        with self.lock:
-            needs_refresh = self.refresh_time < current_time
-            if needs_refresh:
-                self.refresh_time = current_time + 60
-
-        return needs_refresh
-
-    def _save_token_info(self, token_response: dict) -> None:
-        """
-        Decode the access token and save the response from the JWT service to the object's state
-
-        Refresh time is set to approximately 80% of the token's TTL to ensure that
-        the token refresh completes before the current token expires.
-
-        Parameters
-        ----------
-        token_response : dict
-            Response from token service
-        """
-        self.token_info = token_response
+    def extract_exp_and_ttl(self, token_response):
```

Comment:
`extract_exp_and_ttl` function also makes token parsing and storing, so the name of the function does not fully cover its full responsibility.
I would vote to have the `_save_token_info` again as abstract function that should be implement with tokenManager specific methods.

Reply:
I believe this was done in 07136fd526ef48939bbc44233c3cb8a08eeb1b01 . Let me now if I've missed anything there.